src/main/php/xp/runtime/Modules.class.php

```diff
@@ -112,11 +119,39 @@ private function load($namespace, $module, $version= null) {
 
       $this->loaded[$module]= true;
       foreach ($defines['autoload']['files'] ?? [] as $file) {
-        require($base.strtr($file, '/', DIRECTORY_SEPARATOR));
+        require $base.strtr($file, '/', DIRECTORY_SEPARATOR);
+      }
+
+      // See https://www.php-fig.org/psr/psr-0/: Underscores special case, e.g.
+      // name\space\Class_Name => name/space/Class/Name.php
+      foreach ($defines['autoload']['psr-0'] ?? [] as $prefix => $source) {
+        $path= $base.rtrim(strtr($source, '/', DIRECTORY_SEPARATOR), DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR;
```

Comment:
This does not handle arrays as document [here](https://getcomposer.org/doc/04-schema.md#autoload):

```json
{
    "autoload": {
        "psr-4": { "Monolog\\": ["src/", "lib/"] }
    }
}
```